addons/mail/static/src/models/autocomplete_input_view/autocomplete_input_view.js

```diff
@@ -0,0 +1,224 @@
+            }
+        },
+        /**
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computeCustomClass() {
+            if (this.discussSidebarCategoryOwnerAsAddingItemInput) {
+                if (this.discussSidebarCategoryOwnerAsAddingItemInput === this.messaging.discuss.categoryChannel) {
+                    return 'o_DiscussSidebarCategory_newChannelAutocompleteSuggestions';
+                }
+            }
+            if (this.messagingMenuOwnerAsMobileNewMessageInput) {
+                return 'o_MessagingMenu_mobileNewMessageInputAutocomplete';
+            }
+        },
+        /**
+         * @private
+         * @returns {boolean|FieldCommand}
+         */
+        _computeIsHtml() {
+            if (this.discussSidebarCategoryOwnerAsAddingItemInput) {
+                return this.discussSidebarCategoryOwnerAsAddingItemInput === this.messaging.discuss.categoryChannel;
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computePlaceholder() {
+            if (this.chatWindowOwnerAsNewMessageFormInput) {
+                return this.env._t("Search user...");
+            }
+            if (this.discussOwnerAsChannelMobileAddItemHeader) {
+                return this.env._t("Create or search channel...");
+            }
+            if (this.discussOwnerAsChatMobileAddItemHeader) {
+                return this.env._t("Search user...");
+            }
+            if (this.discussSidebarCategoryOwnerAsAddingItemInput) {
+                return this.discussSidebarCategoryOwnerAsAddingItemInput.newItemPlaceholderText;
+            }
+            if (this.messagingMenuOwnerAsMobileNewMessageInput) {
+                return this.env._t("Search user...");
+            }
+            return clear();
+        },
+    },
+    fields: {
+        chatWindowOwnerAsNewMessageFormInput: one('ChatWindow', {
+            inverse: 'newMessageFormInputView',
+            readonly: true,
+        }),
+        /**
+         * States the OWL component of this autocomplete input view.
+         */
+        component: attr(),
+        customClass: attr({
+            compute: '_computeCustomClass',
+            default: '',
+        }),
+        discussOwnerAsChannelMobileAddItemHeader: one('Discuss', {
+            inverse: 'channelMobileAddItemHeaderInputView',
+            readonly: true,
+        }),
+        discussOwnerAsChatMobileAddItemHeader: one('Discuss', {
+            inverse: 'chatMobileAddItemHeaderInputView',
+            readonly: true,
+        }),
+        discussSidebarCategoryOwnerAsAddingItemInput: one('DiscussSidebarCategory', {
+            inverse: 'addingItemInputView',
+            readonly: true,
+        }),
+        doFocus: attr({
+            default: false,
+        }),
+        isFocused: attr({
+            default: false,
+        }),
+        isHtml: attr({
```

Comment:
I would add a comment than when this field is true, we must make sure the result of `onSource` must be properly escaped.